### PR TITLE
hide duplicate org tasks from case timeline and active tasks

### LINKED ISSUE #8937: Hide duplicate organization tasks from case timeline
When we [added task instructions to case timeline](https://github.com/department-of-veterans-affairs/caseflow/issues/8483) we accidentally started showing organization tasks in the case timeline again (these were previously hidden by the work done in #7943). Additionally, since we've decided to [show incomplete tasks that aren't actionable](https://github.com/department-of-veterans-affairs/caseflow/issues/8220) in the task snapshot instead of the case timeline area as recently planned, we should consider moving the construction of the set of tasks to display in the case timeline back to the backend (#8483 moved that logic to the frontend).

image

## Acceptance criteria
* [ ] Do not display duplicate tasks in case timeline

### PR DESCRIPTION
**Pull request by youngfreezyVA**
Resolves #8937 #9785

### CODE REVIEW

File: app/models/task.rb [hunk 1]
```diff
@@ -143,12 +143,16 @@ def hide_from_queue_table_view
     false
   end
 
+  def duplicate_org_task
+    assigned_to.is_a?(Organization) && children.pluck(:assigned_to_type).include?(User.name)
```
Comment (aroltsch) [inline app/models/task.rb]: @lowellrex could you confirm this is correct

File: app/models/task.rb [hunk 2]
```diff
@@ -143,12 +143,16 @@ def hide_from_queue_table_view
...
+  end
+
   def hide_from_case_timeline
-    false
+    duplicate_org_task
   end
 
   def hide_from_task_snapshot
-    false
+    duplicate_org_task
```
Comment (lowellrex) [inline app/models/task.rb]: This is such a clever way to solve this problem! Bravo!

File: app/models/task.rb (hunk 1 above)
Comment (lowellrex) [reply app/models/task.rb]: I think we also need to confirm that the child task that is assigned to a user has the same type as the current task. Maybe something like:
```ruby
assigned_to.is_a?(Organization) && children.any? { |t| User.name == t.assigned_to_type && type == t.type }
```

File: app/models/task.rb (hunk 2 above)
Comment (youngfreezyVA) [reply app/models/task.rb]: thank you!

File: app/models/task.rb [hunk 3]
```diff
@@ -143,12 +143,16 @@ def hide_from_queue_table_view
     false
   end
 
+  def duplicate_org_task
```
Comment (aroltsch) [inline app/models/task.rb]: Let's add a unit test for this method.

Comment (youngfreezyVA) [reply app/models/task.rb]: donesies

**Pull request merged**